docs/releasing/README.md

```diff
@@ -0,0 +1,182 @@
+# Releasing
+
+## When to release
+
+- When a [Subnet-EVM](https://github.com/ava-labs/avalanchego/releases) release is made
+- If there is a significant code change or bugfix in this repository
+
+## Procedure
+
+In this section, we create a release `v0.4.0`. We therefore assign these environment variables to simplify copying instructions:
+
+```bash
+export VERSION=v0.4.0
+export GITHUB_USER=username
+```
+
+`GITHUB_USER` is the username of the person creating the release. This is simply used to prefix branches.
+
+1. Create your branch, usually from the tip of the `main` branch:
+
+    ```bash
+    git fetch origin main:main
+    git checkout main
+    git checkout -b "$GITHUB_USER/releases/$VERSION"
+    ```
+
+1. Modify the [plugin/main.go](../../plugin/main.go) `Version` global string constant and set it to the desired `$VERSION`.
+1. Upgrade the [subnet-evm Go dependency](https://github.com/ava-labs/subnet-evm/releases), which will likely also upgrade the AvalancheGo dependency version:
+
+    ```bash
+    go get github.com/ava-labs/subnet-evm
+    go mod tidy
+    ```
+
+1. Get the Go version from the `go.mod` file:
+
+    ```bash
+    cat go.mod | grep -oE "^go 1\.\d+" | grep -oE "1\.\d+"
```

Comment:
I think we can keep this simple by just taking a look at the go.mod file and keep it manual. I just have a slight concern that making this doc depending on scripts/cmd might result into overcomplicating steps.